### PR TITLE
Feature/best worst offense

### PR DESCRIPTION
**Pull request by TLamb32**
## Description

This PR created the helper methods: #alltime_goals_per_game, #games_played_per_team,  #find_best_offense_team_id, #find_worst_offense_team_id.

These methods add functionality to the league methods: #best_offense and #worst_offense

## Type of change

- [x] fix
- [x] feat
- [x] test
- [x] refactor
- [ ] docs

## Checklist

- [x] code has been self reviewed
- [x] code runs without any errors
- [x] thorough testing has been implemented if adding feature
- [x] all tests pass

### Thanks!

### CODE REVIEW

**Review by aevans27 — APPROVED** [review]
The test setup looks interesting.

File: spec/team_result_parser_spec.rb [hunk 1]
```diff
@@ -27,4 +29,117 @@
...
+  describe "#offense" do
+    it "can find the best and worst offenses" do
+      league_parser = LeagueParser.new
+      team_result_parser = TeamResultParser.new
+
+      team_result_parser.get_game_info
+      league_parser.list_teams
+
+      expect(team_result_parser.best_offense(league_parser.teams_list)).to eq("Reign FC")
+      expect(team_result_parser.worst_offense(league_parser.teams_list)).to eq("Utah Royals FC")
```
Comment (ttakahashi1591) [inline spec/team_result_parser_spec.rb:42]: Not sure if its the view in here or do these two need to be tabbed over?

Comment (lanebret85) [reply spec/team_result_parser_spec.rb:42]: Not sure I'm seeing that. I think it's good!

File: spec/league_parser_spec.rb
Comment (lanebret85) [inline spec/league_parser_spec.rb:1]: nice work removing the unnecessary argument from the LeagueParser objects

File: lib/league_parser.rb
Comment (lanebret85) [inline lib/league_parser.rb:1]: this class looks much better organized now!

**Review by lanebret85 — APPROVED** [review]

**Pull request merged**